app/services/recovery_code_generator.rb

```diff
@@ -1,7 +1,7 @@
 class RecoveryCodeGenerator
   attr_reader :user_access_key
 
-  def initialize(user, length: 16)
+  def initialize(user, length: 12)
```

Comment:
I'm not sure why it was done this way originally, but Figaro.env is used in the `recovery_code_length` method, which is where the actual random phrase length is defined.